test/GPU/Vulkan/Integration/matmul-vulkan.mlir

```diff
@@ -0,0 +1,41 @@
+// RUN: ASAN_OPTIONS=protect_shadow_gap=0:replace_intrin=0:detect_leaks=0:${ASAN_OPTIONS} \
+// RUN: tpp-run %s -gpu=vulkan \
+// RUN:  -entry-point-result=void -e entry 2>&1 | \
+// RUN: FileCheck %s
+
+module attributes {
+  gpu.container_module,
+  spirv.target_env = #spirv.target_env<
+    #spirv.vce<v1.0, [Shader], [SPV_KHR_storage_buffer_storage_class]>, #spirv.resource_limits<>>
+} {
+  func.func @entry(%arg0: memref<8x8xf32>, %arg1: memref<8x8xf32>, %arg2: memref<8x8xf32>) {
+    %c8 = arith.constant 8 : index
+    %c0 = arith.constant 0 : index
+    %c1 = arith.constant 1 : index
+    gpu.launch_func  @entry_kernel::@entry_kernel blocks in (%c8, %c8, %c1) threads in (%c1, %c1, %c1) args(%arg0 : memref<8x8xf32>, %arg1 : memref<8x8xf32>, %arg2 : memref<8x8xf32>, %c0 : index, %c8 : index, %c1 : index)
+    %cast = memref.cast %arg2 : memref<8x8xf32> to memref<*xf32>
+    call @printMemrefF32(%cast) : (memref<*xf32>) -> ()
+    return
+  }
+
+  gpu.module @entry_kernel {
+    gpu.func @entry_kernel(%arg0: memref<8x8xf32>, %arg1: memref<8x8xf32>, %arg2: memref<8x8xf32>, %arg3: index, %arg4: index, %arg5: index)
+    kernel attributes { spirv.entry_point_abi = #spirv.entry_point_abi<workgroup_size = [1, 1, 1]> } {
```

Comment:
If this is comparable to the CUDA matmul example, than this is way more complex. Can we get a similar kernel from CUDA and lower it all the way down to SPIRV via Vulkan and get this IR?

Reply:
It's not meant to be comparable. This test's goal is to have lower level IR at SPIRV(ish) level to test that for loop lowering works properly as previously it was not possible. It's done at the same level as `addf-vulkan.mlir`.

Comparable examples are covered by `linalg-matmul.mlir`, `tpp-gemm.mlir`, and `tpp-brgemm.mlir` tests present in both `CUDA` and `Vulkan`.